lib/request_handler/multipart_parser.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'request_handler/error'
+require 'request_handler/schema_parser'
+require 'request_handler/error'
+require 'request_handler/json_api_data_parser'
+module RequestHandler
+  class MultipartsParser
+    def initialize(request:, multipart_config:)
+      @request = request
+      @params = request.params
+      @multipart_config = multipart_config
+      missing_arguments = []
+      missing_arguments << { params: 'is missing' } if params.nil?
+      missing_arguments << { multipart_config: 'is missing' } if multipart_config.nil?
+      raise MissingArgumentError, missing_arguments unless missing_arguments.empty?
+    end
+
+    def run
+      multipart_config.keys.each_with_object({}) do |name, memo|
+        raise ExternalArgumentError, multipart: 'missing' if params[name.to_s].nil?
+        memo[name] = parse_part(name)
+      end
+    end
+
+    private
+
+    def parse_part(name)
+      if lookup("#{name}.schema")
+        raise ExternalArgumentError, multipart_file: 'missing' if multipart_file(name).nil?
```

Comment:
This check should be moved in front of the if as it should also be done for a "real file".